Sunrise/ViewModels/CartViewModel.swift

```diff
@@ -0,0 +1,189 @@
+    }
+
+    func lineItemImageUrlAtIndexPath(indexPath: NSIndexPath) -> String {
+        return cart.value?.lineItems?[indexPath.row].variant?.images?.first?.url ?? ""
+    }
+
+    func lineItemOldPriceAtIndexPath(indexPath: NSIndexPath) -> String {
+        guard let price = cart.value?.lineItems?[indexPath.row].price, value = price.value,
+        _ = price.discounted?.value else { return "" }
+
+        return value.description
+    }
+
+    func lineItemPriceAtIndexPath(indexPath: NSIndexPath) -> String {
+        guard let price = cart.value?.lineItems?[indexPath.row].price, value = price.value else { return "" }
+
+        if let discounted = price.discounted?.value {
+            return discounted.description
+        } else {
+            return value.description
+        }
+    }
+
+    func lineItemQuantityAtIndexPath(indexPath: NSIndexPath) -> String {
+        return cart.value?.lineItems?[indexPath.row].quantity?.description ?? "0"
+    }
+
+    func lineItemTotalPriceAtIndexPath(indexPath: NSIndexPath) -> String {
+        return cart.value?.lineItems?[indexPath.row].totalPrice?.description ?? "N/A"
+    }
+
+    // MARK: - Commercetools product projections querying
+
+    private func queryForActiveCart() {
+        isLoading.value = true
+
+        // Get the cart with state Active which has the most recent lastModifiedAt.
+        Commercetools.Cart.query(predicates: ["cartState=\"Active\""], sort: ["lastModifiedAt desc"], limit: 1,
+                result: { result in
+                    if let results = result.response?["results"] as? [[String: AnyObject]],
+                    carts = Mapper<Cart>().mapArray(results) where result.isSuccess {
+                        self.cart.value = carts.first
+
+                    } else if let errors = result.errors where result.isFailure {
+                        super.alertMessageObserver.sendNext(self.alertMessageForErrors(errors))
+
+                    }
+                    self.isLoading.value = false
+                })
+    }
+
+    // MARK: - Cart overview calculations
+
+    private func calculateOrderTotal() -> String {
+        guard let cart = cart.value, totalPrice = cart.totalPrice else { return "" }
+
+        if let totalGross = cart.taxedPrice?.totalGross {
+            return totalGross.description
+
+        } else {
+            return totalPrice.description
+        }
+    }
+
+    private func calculateOrderDiscount() -> String {
+        guard let lineItems = cart.value?.lineItems else { return "" }
+
+        let totalOrderDiscountAmount = lineItems.reduce(0, combine: { $0 + calculateCartDiscountForLineItem($1) })
+        return Money(currencyCode: lineItems.first?.totalPrice?.currencyCode ?? "",
+                     centAmount: totalOrderDiscountAmount).description
+
+    }
+
+    private func calculateCartDiscountForLineItem(lineItem: LineItem) -> Int {
```

Comment:
@cneijenhuis @mmoelli - this is the part where I could really use your pair of eyes to double check the discount logic..